Lib/doctest.py

```diff
@@ -1528,6 +1528,8 @@ def out(s):
         self.debugger.reset()
         pdb.set_trace = self.debugger.set_trace
 
+        import linecache
```

Comment:
Not sure there' much point in making this one a lazy import. How many times does someone import `doctest` and then _not_ call this method? It's pretty core to the module's functionality, no?

If there are plausible situations where someone might import `doctest` but not call this method, it's worth it to make it a lazy import. But if it just gets imported anyway as soon as `doctest` is actually _used_, I think it would be better to continue importing it eagerly at the top of the module, as we're doing now.

Reply:
I wrote this change while trying hard to reduce the number of imports in libregrtest. Maybe https://github.com/python/cpython/pull/111884 makes this change less important.

I converted this PR as a draft for now. I will see how it goes with other changes to decide later if this change is "needed" or not.